pkg/virt-api/webhooks/validating-webhook/admitters/vmrestore-admitter.go

```diff
@@ -162,6 +167,54 @@ func (admitter *VMRestoreAdmitter) Admit(ctx context.Context, ar *admissionv1.Ad
 	return &reviewResponse
 }
 
+func (admitter *VMRestoreAdmitter) validateTargetVM(ctx context.Context, field *k8sfield.Path, vmRestore *snapshotv1.VirtualMachineRestore) (causes []metav1.StatusCause, err error) {
+	targetName := vmRestore.Spec.Target.Name
+	namespace := vmRestore.Namespace
+
+	causes = admitter.validatePatches(vmRestore.Spec.Patches, field.Child("patches"))
+
+	vmSnapshot, err := admitter.Client.VirtualMachineSnapshot(namespace).Get(ctx, vmRestore.Spec.VirtualMachineSnapshotName, metav1.GetOptions{})
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return nil, nil
+		}
+		return nil, err
+	}
+
+	target, err := admitter.Client.VirtualMachine(namespace).Get(ctx, targetName, metav1.GetOptions{})
+	if err != nil && !errors.IsNotFound(err) {
+		return nil, err
+	}
+
+	sourceTargetVmsAreDifferent := errors.IsNotFound(err) || (vmSnapshot.Status.SourceUID != nil && target.UID != *vmSnapshot.Status.SourceUID)
+	if sourceTargetVmsAreDifferent {
+		contentName := vmSnapshot.Status.VirtualMachineSnapshotContentName
+		if contentName == nil {
+			return nil, fmt.Errorf("snapshot content name is nil in vmSnapshot status")
+		}
+
+		vmSnapshotContent, err := admitter.Client.VirtualMachineSnapshotContent(namespace).Get(ctx, *contentName, metav1.GetOptions{})
+		if err != nil {
+			return nil, err
+		}
+
+		snapshotVM := vmSnapshotContent.Spec.Source.VirtualMachine
```

Comment:
Just noticed that these snapshot checks kinda defeat the purpose of https://github.com/kubevirt/kubevirt/pull/13305. @ShellyKa13 I'll open a follow-up to fully address this part in the controller once the online snapshot support merges.